### DISCUSSION TITLE
v1.4.2

### DISCUSSION THREAD

**Discussion by eduardofilo** [eduardofilo/RG350_adam_image#213] [Announcements]
## Change log

Minor version to fix some problems found in 1.4.1 version since publishing. The changes introduced are:

* Installed SimpleMenu v10.3 that returned to v10.2 in 1.4.1 by mistake. This finally makes the launchers introduced in 1.4.1 in the apps section work.
* Solved issue #210 with SimpleMenu alias.txt file by @neilswann80.
* Improved folder scaffolding script by @neilswann80.

Follow installation/upgrade instructions on 1.4 release description.

The `upgrade_1.4.2.tgz` file allows to update versions 1.4 or 1.4.1 to 1.4.2 without having to flash again. To do this, proceed as follows:

1. Copy the `upgrade_1.4.2.tgz` file to the root of the external card.
2. Insert the external card in the console and boot. 
3. Connect console to PC with USB cable.
4. Open a terminal in Windows or Linux and initiate a SSH session with command `ssh od@10.1.1.2` (the password is in [README>System access](https://github.com/eduardofilo/RG350_adam_image/#system-access)).
5. Untar the file copied in step 1 with this command (same password than in step 4):

    ```
    sudo tar -xzf /media/sdcard/upgrade_1.4.2.tgz -C / && reboot
    ```
    
6. You are done.

<hr /><em>This discussion was created from the release <a href='https://github.com/eduardofilo/RG350_adam_image/releases/tag/v1.4.2'>v1.4.2</a>.</em>

**Comment by jajabinx55** [comment]
My weekend is looking even better. Thanks so much for your amazing work. Will try this out when I get home.

**Reply by eduardofilo** [reply]
This is a minor update. If you already have v1.4 or v1.4.1 installed, review the changes made to v1.4.2 and, if they do not affect you, it may not be worth the risk of losing data by reinstalling.

**Comment by ForceNever** [comment]
Add battery level to main menu please.

**Reply by neilswann80** [reply]
@ForceNever This would need to be changed in [SimpleMenu](https://github.com/fgl82/simplemenu) I believe, so you'd need to raise this with @fgl82

**Reply by aplanas** [reply]
Seems that there is already an issue for tracking this: https://github.com/fgl82/simplemenu/issues/186

**Comment by MrWidge** [comment]
Is there no longer an option for an auto shutdown? I know there's screen time out. But I swear there was a setting previously to shutdown after a set amount of time. Also no option to control brightness apart from the shortcut commands? Using the 280v

**Reply by neilswann80** [reply]
By pressing `START` you should be able to change the `Screen timeout` option.  The default is no timeout with the screen `always on`

For changing brightness, no there does not appear to be a GUI setting or app to do this.  The shortcut commands work great though.  Any reason why you'd want to use anything else?
FYI... shortcut command is `POWER` and `LEFT D-PAD` or ` RIGHT D-PAD`

**Reply by eduardofilo** [reply]
There is also a setting for brightness in GMenu2X in the settings menu of that frontend.
SimpleMenu also includes a brighness settings in their experimental branch. Therefore I suppose that in the future this frontend will also include it.

**Comment by Juanpoldo** [comment]
I install the latest version and notice this error on Disk Genius is there other way to fix this? I want to use the unallocated size for media consumption.

Note: Im using 8 and 16gb card for testing

![FB_IMG_1647128644265](https://user-images.githubusercontent.com/101152609/158038940-5738bcea-f47a-450b-8473-357c01f5c270.jpg)

![FB_IMG_1647128637417](https://user-images.githubusercontent.com/101152609/158038943-76d93d6d-0553-4453-9142-ba121f8ec189.jpg)

**Reply by eduardofilo** [reply]
Just don't use DiskGenius. I know of many cases of ext4 file systems corrupted by that program. The partition is checked and the errors corrected just before the card dump:

https://github.com/eduardofilo/RG350_adam_image/blob/036e82e6e9debe5a744d26f5f31560cfdf8bd84a/manual_build/build.sh#L185

Just in case, I have checked from Linux the partition just after flashing and it didn't detect any errors (sorry for the captures in spanish):

```
edumoreno@eduardo-HP-Folio-13:~$ sudo e2fsck -f -n /dev/mmcblk0p2
e2fsck 1.46.3 (27-Jul-2021)
Paso 1: Verificando nodos-i, bloques y tamaños
Paso 2: Verificando la estructura de directorios
Paso 3: Revisando la conectividad de directorios
Paso 4: Revisando las cuentas de referencia
Paso 5: Revisando el resumen de información de grupos
/dev/mmcblk0p2: 3510/179168 ficheros (1.8% no contiguos), 178410/716800 bloques
edumoreno@eduardo-HP-Folio-13:~$ echo $?
0
```

The `0` at the end correspond to:

```
EXIT CODE
       The exit code returned by e2fsck is the sum of the following conditions:
            0    - No errors
            1    - File system errors corrected
            2    - File system errors corrected, system should
                   be rebooted
            4    - File system errors left uncorrected
            8    - Operational error
            16   - Usage or syntax error
            32   - E2fsck canceled by user request
            128  - Shared library error
```

Using gparted in Linux the partitions resizes without problems:

![gparted](https://user-images.githubusercontent.com/2923728/158057077-9feadd0f-185e-4d78-9d31-ae045d04c7f6.png)

**Reply by Juanpoldo** [reply]
What app should i use aside from disk genius? Im using a windows PC

**Reply by eduardofilo** [reply]
I don't use Windows much, but DiskGenius seems to be one of the best and as you can see it gives problems. There doesn't seem to be much market in the Windows environment to manage ext4 partitions. Maybe installing WSL on Windows there are more options, but I don't know, sorry. The only option I can think of is to use a LiveUSB with some Linux distribution to resize the partition.

**Reply by neilswann80** [reply]
I'd recommend this [LiveUSB](https://www.system-rescue.org/Download/)

Smaller download and a really useful tool in its own right.  Will boot to a CLI, just type `startx` to then boot a GUI and you can use `gparted` to then adjust the partition.

You may have to enable `Legacy boot devices` in your PCs BIOS to boot this USB.

**Comment by Strudell-code** [comment]
Thank you for the releases! Just a quick feedback that I noticed... I use a 350P device and play Tekken 3 with the overclock option. Since version 1.4 the overclock option trying to start the game results in a crash followed by a reset. I tried version 1.4, 1.4.1 and 1.4.2 and all are with the same issue. Last version in which it is working as of now is the 1.3.1

**Reply by eduardofilo** [reply]
Thank you for the report.
The 1.4.x versions share the system which is where the OC is managed. It makes no sense to roll back the version just because of a problem like that, so I'm afraid you'll have to live with it. Also the possibility to use OC is particular to each chip, so it might work for other users. I for example can't OC either of my two machines. Since the first revision.

**Reply by eduardofilo** [reply]
It could also be that a future version of the system will change/improve the situation on your machine. There is actually a new version of ODbeta later than the one installed in version 1.4.2 of Adam. You can try to update manually and see if you have any luck. I don't think it changed anything that has any influence on this issue though.
http://od.abstraction.se/opendingux/latest/gcw0-update-2022-02-13.opk

**Reply by Strudell-code** [reply]
Thank you for taking the time to check into it and give me a (very quick!) feedback! I'm learning my way on the user side of this... tried but could not update the ODbeta using the opk launching it on commander or by OS update (I'm sure this is just me in need to research further on how to do it properly). Anyway I'll keep an eye as always on your new releases and will comment if the OC option comes back to work on my chip in a future release.

**Reply by Strudell-code** [reply]
quick update! managed to update with the provided ODbeta and tested it again. Can confirm that the OC issue remain. Will further test on later releases
